### PR TITLE
VACMS-8048: Allow authenticated users to export CSV of users

### LINKED ISSUE #8048: Allow auth users to export CSV of users
## User Story or Problem Statement

As a business partner such as VHA digital media, i want to know who has access to edit content for my products (like, has a specific role), so that i can review for governance permissions. 

## Acceptance Criteria
- [ ] Non admin users can get CSV reports like this one`prod.cms.va.gov/admin/people/vacms-user-export.csv?user=&status=1&role=vamc_content_creator&workbench_access_section__section=All&page&_format=csv`

## Implementation steps

<details><summary>This approach failed.</summary>

We haven't created product-specific account admin roles, or built out reporting tools (like the Section members tab), Ii think we can solve for this by simply extending access to CSV dumps of /admin/people to authenticated users. 

1. Go to /admin/structure/views/view/user_admin_people/edit/data_export_1
2. Override the display's permission setting (but not the default permission setting) to Role, and make it available to any authenticated users
3. Test `prod.cms.va.gov/admin/people/vacms-user-export.csv?user=&status=1&role=vamc_content_creator&workbench_access_section__section=All&page&_format=csv` as any non-admin user. You should get a download. 

This failed, probably because the user probably needs access to the view with which the CSV export is generated.

</details>

Workaround: 
* CMS help desk team bookmarks [this CSV export link](prod.cms.va.gov/admin/people/vacms-user-export.csv?user=&status=1&role=vamc_content_creator&workbench_access_section__section=All&page&_format=csv) and runs this report periodically on behalf of the users. 

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [x] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team
Please check the team(s) that will do this work. 

- [ ] `CMS Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team ` (leave Sitewide unchecked and check the specific team instead)
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by netboss**
- Created the new CSV display and configure it:
    - Removed "section" filter
    - Adjusted permission to be by role authenticated user
    - Updated the CSV display path to /admin/people/vacms-all-users-export.csv
    - Added the footer to the people main display (which later remove because the CSV icon was showing twice)
- Updated the Drupal spec tool and then updated the behat test.

## Description

closes #8048 

## Testing done
Passes all tests

## QA steps

As any authenticated user, content creator, content editor, content reviewer, content publisher, content admin and administrator

- [x] Log-in to the CMS and go to https://pr8250-1wt1r0qcw1cajbgtbtamv3h1eazkinzz.ci.cms.va.gov/admin/people/vacms-all-users-export.csv?user=&status=1&role=All&page&_format=csv to verify and confirm that the CSV download is accessible for all the above user roles validate that you get a download
    - [x] validate that you get a csv all users.
- [x]   Visit https://pr8250-1wt1r0qcw1cajbgtbtamv3h1eazkinzz.ci.cms.va.gov/admin/people/vacms-all-users-export.csv?user=&status=1&role=content_admin&page&_format=csv  validate that you get a download
   - [x]  Validate that the CSV only contains content admins. 
- [x] Logged-in as an administrator go to `/admin/people` to verify and confirm that there is a "Download CSV of all users" button in the view header just above the filters

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
I think we want to expand these QA steps out for a couple different user rolls to make sure it works for everyone.

We might also add some test steps for different variations on the filters that were requested to support in the issue
image

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This default treatment from the Views is pretty confusing
image

I like the pattern @msbtterswrth has established with putting something like this in the top of the view header resulting in a look like this.
image

```
<div><a href="/admin/people/vacms-all-users-export.csv" class="button">Download CSV of all users</a></div>
```
And then change the "Attach to:" be none


I think a couple things got misconfigured, see below.

File: config/sync/views.view.user_admin_people.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ langcode: en
 status: true
 dependencies:
   config:
+    - user.role.authenticated
```
[the remark this answers is not in the document]
Comment (netboss) [reply config/sync/views.view.user_admin_people.yml:6]: @swirtSJW 
Weird, in my local this is correctly set. Both the page and data_export_1 displays have their permissions set to: `Access: Permission | Administer users`.

File: config/sync/views.view.user_admin_people.yml [hunk 2]
```diff
@@ -972,9 +973,10 @@ display:
           entity_type: user
           plugin_id: workbench_access_section
           operator: in
-          value: {  }
+          value:
+            - All
           group: 1
-          exposed: true
```
[the remark this answers is not in the document]
Comment (netboss) [reply config/sync/views.view.user_admin_people.yml:977]: @swirtSJW 
Ah, I see the problem. The "Expose this filter to visitors, to allow them to change it" checkbox was unchecked for both the page and data_export_1 displays. It must have been when I was in the process of removing it from the new data_export_2 where I first tried "hiding" it by unchecking the above checkbox (and probably did that for all displays). Fixed it. Will push soon.

File: config/sync/views.view.user_admin_people.yml [hunk 3]
```diff
@@ -1130,7 +1132,7 @@ display:
         type: views_query
         options:
           query_comment: ''
-          disable_sql_rewrite: false
+          disable_sql_rewrite: true
```
[the remark this answers is not in the document]
Comment (netboss) [reply config/sync/views.view.user_admin_people.yml:1135]: @swirtSJW 
This was my mistake. I thought I did this to the data_export_2 display only. My bad. Fixed.

**Comment by swirtSJW** [comment]
There are a couple related patches that allow for differentiating the duplicate download buttons.
https://www.drupal.org/project/views_data_export/issues/3200051
https://www.drupal.org/project/views_data_export/issues/2943895

**Comment by netboss** [comment]
> There are a couple related patches that allow for differentiating the duplicate download buttons. https://www.drupal.org/project/views_data_export/issues/3200051 https://www.drupal.org/project/views_data_export/issues/2943895

@swirtSJW 
I already added the CSV download button to the view's header. See screen shot:
VACMS-8048_people_new_CSV_download_button

Do we want to use the above or use one of the two available patches?

**Comment by swirtSJW** [comment]
@netboss  If there are new things to test (like the functioning of a new button) lets add specific QA steps to make sure things are working.

**Comment by swirtSJW** [comment]
I am not seeing the new button on this PR environment.
I am only seeing 
image


and 

image

Did you push your latest changes?

**Comment by swirtSJW** [comment]
@kevwalsh I confirmed that as any authed user (like RS) I am able to access the download directly if I am provided a link 
https://pr8250-1wt1r0qcw1cajbgtbtamv3h1eazkinzz.ci.cms.va.gov/admin/people/vacms-all-users-export.csv?page&_format=csv

The button is visible to anyone who can access admin/people but that set of users is limited by role.

image

**Review by swirtSJW — DISMISSED** [review]

**Review by kevwalsh — DISMISSED** [review]
We should be able to specify the drupal role as a parameter in the csv download.

**Comment by swirtSJW** [comment]
@netboss  It looks like your force push confused tugboat again and it stalled out.  
I just rebuilt it to get things started up again.
image

**Review by swirtSJW — APPROVED** [review]
QA checks out and includes the role filter query param.

**Pull request merged**